AK/IPv6Address.h

```diff
@@ -256,13 +257,88 @@ class [[gnu::packed]] IPv6Address {
         return IPv6Address({ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 });
     }
 
+    [[nodiscard]] constexpr bool is_loopback() const
+    {
+        return *this == loopback();
+    }
+
+    [[nodiscard]] constexpr bool is_in_subnet(IPv6Address subnet, u16 network_size) const
+    {
+        VERIFY(network_size <= 128);
+        return this->network(network_size) == subnet;
+    }
+
+    [[nodiscard]] constexpr IPv6Address network(u16 network_size) const
+    {
+        VERIFY(network_size <= 128);
+        u64 const low_word_size = min(network_size, 64);
+        u64 const high_word_size = clamp(static_cast<i16>(network_size) - 64, 0, 64);
+        // FIXME: somehow avoid branching for special case *without* introducing UB?
+        // Bitmagic works as follows:
+        //  - Invert the network size since we can more easily create many mask bits towards the "high" side of the address (i.e. low bits)
```

Comment:
Do we need to be clever here? Doesn't the more straightforward

```
        IPv6Address net;
        for (int i = 0; i < 16; ++i) {
            if (network_size >= 8) {
                net.m_data[i] = m_data[i];
                network_size -= 8;
            } else {
                u8 mask = ((1 << network_size) - 1) << (8 - network_size);
                net.m_data[i] = m_data[i] & mask;
                break;
            }
        }
        return net;
```

work too? (https://godbolt.org/z/G7KzYWTxT)

(Don't need the new private methods then.)

Reply:
It does work, but the point of this was to be compact and fast. (And the bitmasking shenanigans are still necessary.....)

Reply:
I doubt it's slower, it's half the code, and you don't have to think about endianness. Let's do the simpler version.